simdpp/detail/insn/conv_any_to_float32.h

```diff
@@ -168,16 +168,15 @@ float32<4> i_to_float32(const uint32<4>& a)
     }
     return r;
 #elif SIMDPP_USE_SSE2
-    mask_int32<4> is_large = cmp_gt(a, 0x7fffffff);
+    mask_float32<4> is_large = bit_cast<mask_float32<4>>(cmp_lt(int32<4>(a.eval()), 0));	// true when a is in the range [0x80000000, 0xffffffff)
```

Comment:
Just a little nitpick: you can just write `a` here instead of `a.eval()` since `a` is not an expression and even if it was, `cmp_lt` supports vector expressions as arguments.

Reply:
no doubt some leftover from experimentation. Maybe some earlier form of int32<> casting I was using while experimenting required it. Actually I thought you'd have some shorthand for the mask cast.